### DISCUSSION TITLE
Add PhysX to a multi platform CMake project

### DISCUSSION THREAD

**Discussion by frabera** [NVIDIA-Omniverse/PhysX#89] [Q&A] [answered]
I am setting up a CMake project for a multiplatform app (Windows and Linux x64), what is the best way to add PhysX to the project? Do I need to build manually the library and then mantain two different projects?

**Comment by PcChip** [comment]
I'm in the same situation and looking for advice with this as well

**Comment by ayoub-belarbi** [comment]
PhysX have support for both Windows and Linux. We don't provide precompiled PhysX so you need to compile it from source. This involves including in your project and linking PhysX Libraries to your targets etc, Use CMake’s platform checks to handle different compiler flags, preprocessor definitions, and include directories needed for Windows and Linux conditionally. You can refer to PhysX's cmake files to understand more or less how it does support multiple platforms and to configure your project.
So you don't need to maintain two different projects.

Thanks

**Reply by PcChip** [reply]
thanks for this info,

do you have it on your radar to create some kind of example like this, that people who are new to cmake can use to just easily "click and deploy"?

I'm coming from Visual Studio on Windows and things were much easier from that side, especially when using vcpkg.  So entering into the world of linux and cmake with physx feels kind of like getting tossed into the deep end!

**Reply by ayoub-belarbi** [reply, marked as answer]
At present, we don't have such a thing as "click and deploy". Integrating PhysX, particularly on multiple platforms, involves a fair understanding of build systems and compilers, which CMake manages. Given PhysX's complexity and the nature of custom setups it often requires, having a basic knowledge of how CMake and compilers work is essential.
